src-docs/src/views/datagrid/datagrid_example.js

```diff
@@ -39,92 +261,258 @@ export const DataGridExample = {
+      },
+      demo: (
+        <Fragment>
+          <DataGrid />
+          <EuiSpacer size="xxl" />
+          <EuiText>
+            <h2>Snippet with everything on</h2>
+            <p>
+              Here is a very simple data grid example meant to give you an idea
+              of the data structure and callbacks you&apos;ll need to provide.
+            </p>
+          </EuiText>
+          <EuiSpacer />
+          <EuiCodeBlock language="javascript" paddingSize="s" isCopyable>
+            {gridSnippet}
+          </EuiCodeBlock>
+          <EuiSpacer size="xl" />
+          <EuiText>
+            <h3>General props explanation</h3>
+            <p>
+              Please check the props tab in the example above for more
+              explanation on the lower level object types. The majority of the
+              types are defined in the{' '}
+              <a
+                href="https://github.com/elastic/eui/tree/master/src/components/data_grid/data_grid_types.ts"
+                target="_blank">
+                /data_grid/data_grid_types.ts
+              </a>{' '}
+              file.
+            </p>
+          </EuiText>
+          <EuiSpacer />
+          <EuiDescriptionList
+            compressed
+            listItems={gridConcepts}
+            type="responsiveColumn"
+            titleProps={{ style: { width: '20%' } }}
+            descriptionProps={{ style: { width: '80%' } }}
+          />
+        </Fragment>
+      ),
     },
     {
       source: [
         {
           type: GuideSectionTypes.JS,
-          code: dataGridContainerSource,
+          code: inMemoryDataGridSource,
         },
         {
           type: GuideSectionTypes.HTML,
-          code: dataGridContainerHtml,
+          code: inMemoryDataGridHtml,
         },
       ],
-      title: 'Data grid adapts to its container',
+      title: 'In memory settings',
       text: (
-        <p>
-          When wrapped inside a container, like a dashboard panel, the grid will
-          start hiding controls and adopt a more strict flex layout
-        </p>
+        <Fragment>
+          <p>
+            The grid has levels of <strong>in memory</strong> settings that can
+            be set. The higher levels open up more features, but require you to
+            utilize callbacks for sorting and pagination as those features are
+            enabled. The following settings are available. When possible, it is
+            advisable to utilize in memory as deeply as possible to provide a
+            better core experience for the user.
+          </p>
+          <ul>
+            <li>
+              <strong>undefined</strong>: when not in use the grid will not
+              autodetect schemas. Unless you manually provide schemas, sorting
+              will also be heavily limited and likely inaccurate.
+            </li>
+            <li>
+              <strong>enchancements</strong>: provides no in memory operations.
```

Comment:
Enchantments would be super cute but I think right now we're using enhancements

```suggestion
              <strong>enhancements</strong>: provides no in memory operations.
```